### PR TITLE
Bsoptei append user data

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
@adamgyulavari /login and /register endpoints are ready to go (needs fine tuning but ready to connect with front end)

File: src/main/java/com/greenfoxacademy/service/ContactService.java [hunk 1]
```diff
@@ -25,25 +25,29 @@ public ContactService(UserService userService, ContactRepository contactReposito
...
-                getContext().
-                getAuthentication().
-                getName();
-    }
+//    private String obtainUserNameFromSecurity() {
+//        return SecurityContextHolder.
+//                getContext().
+//                getAuthentication().
+//                getName();
+//    }
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/service/ContactService.java]: @bsoptei remove commented code

File: src/main/java/com/greenfoxacademy/responses/BadRequestResponse.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Created by BSoptei on 2/1/2017.
+ */
+public class BadRequestResponse extends CustomServerResponse {
+
+    @Override
+    public ResponseEntity generateResponse() {
+        return new ResponseEntity<>("", createResponseHeaders(), HttpStatus.BAD_REQUEST);
+    }
+}
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/responses/BadRequestResponse.java]: @bsoptei why different classes for creating a ResponseEntity?
the ResponseEntity should be created in the controller, just the first parameters should be defined as different response objects: eg. `ContactListResponse`, `EmptyResponse`, `SingleContactResponse`, etc...
These should be simple classes that can be serialized by the default serializer.

File: src/main/java/com/greenfoxacademy/responses/ListedItemsResponse.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.greenfoxacademy.responses;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+/**
+ * Created by BSoptei on 2/1/2017.
+ */
+public class ListedItemsResponse extends CustomServerResponse {
+    private String body;
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/responses/ListedItemsResponse.java]: @bsoptei not string body here, but a `List<Contact>`

File: src/main/java/com/greenfoxacademy/responses/SuccessfulLoginAndRegistrationResponse.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        return new ResponseEntity<>(body, createResponseHeaders(), HttpStatus.CREATED);
+    }
+
+    @Override
+    protected HttpHeaders createResponseHeaders() {
+        HttpHeaders responseHeaders = new HttpHeaders();
+        URI location = URI.create("https://raptor-konnekt.herokuapp.com");
+        responseHeaders.setLocation(location);
+        responseHeaders.set("session_token",token);
+        return responseHeaders;
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/responses/SuccessfulLoginAndRegistrationResponse.java]: @bsoptei you can call the defaults using `super.createResponseHeaders()`

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: src/main/java/com/greenfoxacademy/service/ContactService.java (hunk 1 above)
Comment (ghost) [reply src/main/java/com/greenfoxacademy/service/ContactService.java]: ok, done

File: src/main/java/com/greenfoxacademy/responses/ListedItemsResponse.java (hunk 3 above)
Comment (ghost) [reply src/main/java/com/greenfoxacademy/responses/ListedItemsResponse.java]: currently it is converted into a String from a List<Contact> (this is one type of a solution for the do-not-display-all-user-data problem)

File: src/main/java/com/greenfoxacademy/responses/SuccessfulLoginAndRegistrationResponse.java (hunk 4 above)
Comment (ghost) [reply src/main/java/com/greenfoxacademy/responses/SuccessfulLoginAndRegistrationResponse.java]: I tried but it won't allow me to 
call  the responseHeaders.set("session_token",token) 
I was going to do it with the getHeaders().set("session_token", token) approach but it's not allowed

File: src/main/java/com/greenfoxacademy/responses/ListedItemsResponse.java (hunk 3 above)
Comment (ghost) [reply src/main/java/com/greenfoxacademy/responses/ListedItemsResponse.java]: btw. we also tried with the @JsonIgnore annotation in the User class but it did not work

File: src/main/java/com/greenfoxacademy/responses/BadRequestResponse.java (hunk 2 above)
Comment (ghost) [reply src/main/java/com/greenfoxacademy/responses/BadRequestResponse.java]: ok, fixed it,
crud responses need fine tuning though.
but /login and /register are cool
going to push soon

**Pull request closed without merging**